### PR TITLE
S3 delete file delete/create dir

### PR DESCRIPTION
**Pull request by vnghia**
@mihaimaruseac 
This PR adds delete/create dir and delete file for s3.

### CODE REVIEW

**Review by mihaimaruseac — APPROVED** [review]

**Review by shivaylamba — APPROVED** [review]

**Review by mihaimaruseac — APPROVED** [review]

**Comment by mihaimaruseac** [comment]
@shivaylamba Respectfully, I'd ask to refrain from approving PRs when the PR was already approved and the review you submitted has no power to trigger the CI/merge (as you are not an owner of the code / in the organisation). Note the grayed checkmark:
pr

Your review in this case blocked CI and there was a need for another review to unblock. Please don't approve random PRs on GitHub.

**Pull request merged**
